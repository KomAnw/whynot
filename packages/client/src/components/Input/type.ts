import { HTMLInputTypeAttribute } from 'react';
import { FieldValues, UseFormRegister } from 'react-hook-form';

export type InputProps = {
  name: string;
  type: HTMLInputTypeAttribute;
  register: UseFormRegister<FieldValues>;
  validationRules: Record<string, unknown>;
  label: string;
  placeholder: string;
  errorMessage?: string;
};
