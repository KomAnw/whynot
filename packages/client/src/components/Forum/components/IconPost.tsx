export const IconPost = () => (
  <svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
    <path
      d="M10.67 19.9701C10.43 19.9901 10.19 20.0001 9.95 20.0001C8.78 20.0001 7.66999 19.7201 6.67999
      19.2301L2.78999 19.7901L3.38 15.8801H3.39C2.99 15.0001 2.75 14.0301 2.75 13.0001C2.75 11.1101 3.51999
      9.40014 4.75999 8.14014"
      stroke="#6457B8"
      strokeWidth="1.5"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
    <path
      d="M10.8601 11.7802C11.3019 11.7802 11.6601 11.431 11.6601 11.0002C11.6601 10.5694 11.3019 10.2202
      10.8601 10.2202C10.4182 10.2202 10.0601 10.5694 10.0601 11.0002C10.0601 11.431 10.4182 11.7802 10.8601 11.7802Z"
      fill="#6457B8"
    />
    <path
      d="M14.0603 11.7802C14.5021 11.7802 14.8603 11.431 14.8603 11.0002C14.8603 10.5694 14.5021 10.2202
      14.0603 10.2202C13.6184 10.2202 13.2603 10.5694 13.2603 11.0002C13.2603 11.431 13.6184 11.7802 14.0603
      11.7802Z"
      fill="#6457B8"
    />
    <path
      d="M17.2502 11.7802C17.692 11.7802 18.0502 11.431 18.0502 11.0002C18.0502 10.5694 17.692 10.2202
      17.2502 10.2202C16.8084 10.2202 16.4502 10.5694 16.4502 11.0002C16.4502 11.431 16.8084 11.7802 17.2502 11.7802Z"
      fill="#6457B8"
    />
    <path
      d="M20.6199 13.88L21.2099 17.79L17.3199 17.23C16.6199 17.58 15.8599 17.82 15.0499 17.93C14.7299
      17.98 14.3899 18 14.0499 18C10.0699 18 6.85986 14.87 6.85986 11C6.85986 9.07 7.64986 7.35 8.94986 6.08C10.2399
      4.79 12.0499 4 14.0599 4C18.0399 4 21.2499 7.13 21.2499 11C21.2499 12.03 21.0099 13 20.6099 13.88H20.6199V13.88Z"
      stroke="#6457B8"
      strokeWidth="1.5"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </svg>
);
