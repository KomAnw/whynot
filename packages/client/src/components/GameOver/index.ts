export { GameOver as default } from './GameOver';
