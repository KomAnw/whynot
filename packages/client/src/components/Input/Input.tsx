import styled from 'styled-components';
import { InputHTMLAttributes } from 'react';
import { InputProps } from 'components/Input/type';
import { Label } from 'src/design/Label';
import { LinkText } from 'src/design/LinkText';
import { breakpoints } from 'src/App/constants';

const { mobileM } = breakpoints;

export const Input = ({ name, type, label, errorMessage, placeholder, validationRules, register }: InputProps) => {
  return (
    <InputContainer>
      <LabelStyled>{label}</LabelStyled>
      <InputStyled type={type} placeholder={placeholder} {...register(name, { ...validationRules })} />
      <ValidationText>{errorMessage}</ValidationText>
    </InputContainer>
  );
};

const InputContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  width: 100%;
  margin: 0;
  font-family: ${({ theme }) => theme.fonts.main};
  font-style: normal;
`;

const ValidationText = styled(Label)`
  margin: 0 auto;
  height: 20px;
  display: block;
  align-items: flex-start;
  color: ${({ theme }) => theme.colors.core.text.error};
`;

const LabelStyled = styled(Label)`
  height: 20px;
  margin: 0;
  display: block;
  align-items: flex-start;
  color: ${({ theme }) => theme.colors.control.input.label};
`;

const InputStyled = styled(LinkText).attrs({ as: 'input' })<InputHTMLAttributes<HTMLInputElement>>`
  box-sizing: border-box;
  border-radius: 5px;
  padding-left: 26px;
  width: 100%;
  height: 40px;
  outline: none;
  background-color: ${({ theme }) => theme.colors.control.input.background};
  border: none;
  color: ${({ theme }) => theme.colors.control.input.color};

  &::placeholder {
    color: ${({ theme }) => theme.colors.control.input.placeHolder};
    font-size: 20px;
    font-weight: 500;
  }

  @media (max-width: ${mobileM}) {
    padding-left: 12px;
  }
`;
