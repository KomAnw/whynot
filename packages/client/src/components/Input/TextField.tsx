import styled from 'styled-components';
import TextField from '@mui/material/TextField';


export const TextField1 = () => {
    return (
      <>
        <Text>Подготовка копонента Input</Text>
        <Text>
          <TextField
            id="outlined-basic"
            label="Outlined"
            variant="outlined"
          />
        </Text>
      </>
    )
}

const Text = styled.p`
  color: blue;
  width: fit-content;
  margin: 100px auto;
`
