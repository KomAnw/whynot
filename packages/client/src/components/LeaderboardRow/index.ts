export { LeaderboardRow as default } from './LeaderboardRow';
