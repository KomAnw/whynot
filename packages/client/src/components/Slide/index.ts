export { SlideHistory, SlideTutorial } from './Slide';
