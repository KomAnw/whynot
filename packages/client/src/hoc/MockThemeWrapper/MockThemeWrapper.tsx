import { ThemeProvider } from 'styled-components';
import { themes } from 'components/App/constants';
import { ThemeWrapperProps } from './types';

const MockThemeWrapper = ({ children }: ThemeWrapperProps) => {
  return <ThemeProvider theme={themes.default}>{children}</ThemeProvider>;
};

export default MockThemeWrapper;
