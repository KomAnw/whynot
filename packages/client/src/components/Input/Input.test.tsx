import { Input } from 'components/Input';
import { create } from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import MockThemeWrapper from 'src/hoc/MockThemeWrapper';
import { Provider } from 'react-redux';
import store from 'src/store';

describe('Input', () => {
  it('renders <Input /> correctly', () => {
    const mockValidationRules = {};
    const mockFn = jest.fn().mockImplementationOnce(() => null);
    const tree = create(
      <Provider store={store}>
        <BrowserRouter>
          <MockThemeWrapper>
            <Input
              key="name"
              register={mockFn}
              name="name"
              type="input"
              label="label"
              placeholder="placeholder"
              validationRules={mockValidationRules}
            />
          </MockThemeWrapper>
        </BrowserRouter>
      </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
