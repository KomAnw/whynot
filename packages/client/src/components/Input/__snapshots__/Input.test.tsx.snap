// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders <Input /> correctly 1`] = `
<div
  className="sc-eDvSVe jOcgQw sc-hLBbgP"
>
  <p
    className="sc-bcXHqe sc-gswNZR sc-gKPRtg bZdROF gQyYxG feiruF"
  >
    label
  </p>
  <input
    className="sc-bcXHqe sc-dkrFOg sc-iBYQkv bZdROF idXDLE haJkzM"
    placeholder="placeholder"
    type="input"
  />
  <p
    className="sc-bcXHqe sc-gswNZR sc-jSUZER bZdROF gQyYxG gurojQ"
  />
</div>
`;
